filebeat/crawler/prospector.go

```diff
@@ -110,6 +110,9 @@ func (p *Prospector) Run(wg *sync.WaitGroup) {
 			logp.Info("Run prospector")
 			p.prospectorer.Run()
 		}
+
+		// Defer next run for the defined scanFrequency
+		time.Sleep(p.ProspectorConfig.ScanFrequencyDuration)
```

Comment:
what's change so we have to delay next run?

Why not:

```
for {
  select {
    case <-p.done:
        ...
    case <-time.After(p.ProspectorConfig.ScanFrequencyDuration):
        p.prospector.Run()
  }
```
?

Problem with time.Sleep is, if it's too long you have to wait for up to scan frequency seconds before potential return from loop if p.done is closed.